src/Components/WebAssembly/WebAssembly/src/Services/LazyAssemblyLoader.cs

```diff
@@ -20,19 +19,22 @@ namespace Microsoft.AspNetCore.Components.WebAssembly.Services
     ///
     /// Supports finding pre-loaded assemblies in a server or pre-rendering context.
     /// </summary>
-    public class LazyAssemblyLoader
+    public sealed class LazyAssemblyLoader
     {
         internal const string GetDynamicAssemblies = "window.Blazor._internal.getLazyAssemblies";
         internal const string ReadDynamicAssemblies = "window.Blazor._internal.readLazyAssemblies";
 
-        private List<Assembly> _loadedAssemblyCache = new List<Assembly>();
+        private readonly IJSRuntime _jsRuntime;
+        private readonly Dictionary<string, Assembly> _loadedAssemblyCache;
 
-        private readonly IServiceProvider _provider;
-
-        public LazyAssemblyLoader(IServiceProvider provider)
+        public LazyAssemblyLoader(IJSRuntime jsRuntime)
         {
-            _provider = provider;
-            _loadedAssemblyCache = AppDomain.CurrentDomain.GetAssemblies().ToList();
+            _jsRuntime = jsRuntime;
+            _loadedAssemblyCache = new Dictionary<string, Assembly>(StringComparer.OrdinalIgnoreCase);
+            foreach (var assembly in AppDomain.CurrentDomain.GetAssemblies())
+            {
+                _loadedAssemblyCache[assembly.GetName().Name + ".dll"] = assembly;
+            }
```

Comment:
```C#
_loadedAssemblyCache = AppDomain.CurrentDomain.GetAssemblies().ToDictionary(a => a.GetName().Name + ".dll");
```

Reply:
You're right about this! I thought I was using the assembly value somewhere but must've been imagining this.